### PR TITLE
Fix restored pvc does not have label issue on guest cluster

### PR DESCRIPTION
**Pull request by xinyanw409**
Signed-off-by: wxinyan <wxinyan@wxinyan-a01.vmware.com>

**What this PR does / why we need it**:
PVC has labels before backup and restore, after restore, new created pvc loses labels. This issue can only be reproduced on guest cluster. 

**Which issue(s) this PR fixes**:
Fix the restored pvc does not have labels issue.
**Special notes for your reviewer**:
Manually tested on guest cluster:
```
# Backup pvc
velero backup create pvc-test-1 --include-namespaces=pvc-test
# Delete pvc
kubectl delete ns pvc-test
# Restore pvc
velero restore create --from-backup pvc-test-1
```
Precheck: https://container-dp.svc.eng.vmware.com/view/CNS-DP/job/Velero-Pipeline-WCP/439/

**Does this PR introduce a user-facing change?**:

```release-note
Fix restored pvc does not have label issue on guest cluster.
```

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/502/

File: pkg/paravirt/paravirt_protected_entity_type_manager.go [hunk 1]
```diff
@@ -427,6 +428,7 @@ func (this *ParaVirtProtectedEntityTypeManager) createGuestPVC(ctx context.Conte
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: gcPVCNamespace,
 			Name:      gcPVCName,
+			Labels:    gcPVCLabels,
```
Comment (xing-yang) [inline pkg/paravirt/paravirt_protected_entity_type_manager.go:431]: What happened to svcPVC.Labels?  Do you see labels on Supervisor Cluster PVC?

With and Without your fix, can you do the following steps and save them in a doc?
1. Create a PVC in Guest Cluster with labels.
2. Check if labels are in both Guest and Supervisor Cluster PVC. Save "kubectl get pvc -o yaml".
3. Check vSphere UI Container Volumes and see if labels are there.  Provide a screen shot.
4. Backup the namespace including the PVC.
4. Delete the namespace from Guest Cluster.  Make sure PVC is deleted from Guest and Supervisor.
5. Check there are no Container Volumes in vSphere.  Take a screen shot.
6. Do a restore.  Save "kubectl get pvc -o yaml" from both Guest and Supervisor.
7. Take a screen shot of container volumes in vSphere.

**Comment by xing-yang** [comment]
The doc you provided in the PR description only contains information for Vanilla cluster, but this bug is specific for Guest Cluster.

File: pkg/paravirt/paravirt_protected_entity_type_manager.go (hunk 1 above)
Comment (xinyanw409) [reply pkg/paravirt/paravirt_protected_entity_type_manager.go:431]: Updated documentation with records for each step.

File: pkg/paravirt/paravirt_protected_entity_type_manager.go [hunk 2]
```diff
@@ -323,19 +323,20 @@ func (this *ParaVirtProtectedEntityTypeManager) getSuperPVCandGuestPVCName(metad
...
 		// NOTE: svcPVC is marshaled from metadata which is originally
 		// from the Guest Cluster PVC
-		return nil, "", "", errors.Wrapf(err, "could not generate PVC name in the Supervisor Cluster for Guest Cluster PVC %s/%s",
+		return nil, "", "", map[string]string{}, errors.Wrapf(err, "could not generate PVC name in the Supervisor Cluster for Guest Cluster PVC %s/%s",
 			svcPVC.Name, svcPVC.Namespace)
 	}
 	// Save original Guest Cluster PVC name and namespace
 	gcPVCName := svcPVC.Name
 	gcPVCNamespace := svcPVC.Namespace
+	gcPVCLabel := svcPVC.Labels
```
Comment (xing-yang) [inline pkg/paravirt/paravirt_protected_entity_type_manager.go:339]: vSphere CSI Driver does not keep Labels on Supervisor PVC.  We should do the same during restore.
Add a line after this to clear out Labels from svcPVC and add comments to the change.

**Review by lintongj — APPROVED** [review]
/lgtm

File: pkg/paravirt/paravirt_protected_entity_type_manager.go (hunk 2 above)
Comment (xinyanw409) [reply pkg/paravirt/paravirt_protected_entity_type_manager.go]: Clear labels from svcPVC and updated documentation.

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/505/

**Comment by xing-yang** [comment]
@xinyanw409 Please add a release note.

**Comment by xing-yang** [comment]
LGTM

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/506/

**Pull request merged**